### PR TITLE
catalog-react: remove Domain column from System table

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Supports #4553 by removing the "Domain" column from the System table presets.  This table is shown on the Domains page, and thus there's no need to repeat showing the domains systems are attached to.

| Before | After |
| --- | --- |
|  image | image |

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4589-gom5rj3jtfalaowywry3rxcq00kpfncs.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602f2881f285af782092bf9e

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4589-gom5rj3jtfalaowywry3rxcq00kpfncs.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602f2881f285af782092bf9e

File: plugins/catalog-react/src/components/EntityTable/presets.tsx [hunk 1]
```diff
@@ -28,7 +27,6 @@ import {
 
 export const systemEntityColumns: TableColumn<SystemEntity>[] = [
   createEntityRefColumn({ defaultKind: 'system' }),
-  createDomainColumn(),
```
Comment (Fox32) [inline plugins/catalog-react/src/components/EntityTable/presets.tsx:31]: I don't thin that it's the right way to remove it from the presets. We should create a custom column layout in the table

**Comment by Fox32** [comment]
What about the other points you made? I think there are also duplicates in the other tables?

**Comment by adamdmharvey** [comment]
Yep - was intending just to take a quick stab at only the domain card as it looked "low hanging fruit", but what you said makes sense. Each display purpose should be able to customize it, for sure.

Think this might be a bit more complicated than I'm able to get to (or know enough about props and prop overrides and such), so for now I'll just close this one out and offer through the issue it as a "help wanted" card for community to chip in!  Hope that makes sense.

**Comment by Fox32** [comment]
Oh, actual this should be to complicated. I can create a PR for it. It's already build to be customizable

**Pull request closed without merging**